lightning-invoice/src/payment.rs

```diff
@@ -715,11 +894,23 @@ mod tests {
 		let payment_id = Some(invoice_payer.pay_invoice(&invoice).unwrap());
 		assert_eq!(*payer.attempts.borrow(), 1);
 
+		{
+			let inflight_map = invoice_payer.create_inflight_map(&payment_hash);
+			// TestRouter returns two hops, with `fee_msat` divided by two. The invoice instantiated above
+			// is hard-coded to test sending an invoice with a value of 128 msats.
+			assert_eq!(inflight_map.get(&(0, true)).unwrap().clone(), 64);
+			assert_eq!(inflight_map.get(&(1, true)).unwrap().clone(), 64);
+		}
+
 		invoice_payer.handle_event(&Event::PaymentSent {
 			payment_id, payment_preimage, payment_hash, fee_paid_msat: None
 		});
 		assert_eq!(*event_handled.borrow(), true);
 		assert_eq!(*payer.attempts.borrow(), 1);
+
+		// We should have cleared the payment_cache of information about this payment
+		let inflight_map = invoice_payer.create_inflight_map(&payment_hash);
+		assert_eq!(inflight_map.len(), 0)
```

Comment:
You'll need to check this prior to handling the `PaymentSent` event because that will clear the entry from the cache entirely. But you want to check that information on a single path is cleared when `PaymentPathSuccessful` or `PaymentPathFailed` is handled. Incidentally, looks like these tests weren't updated when `PaymentPathSuccessful` was added which should be handled prior to `PaymentSent`. Let's make a separate test for this though instead of reusing this one.